financialaid/views.py

```diff
@@ -234,3 +223,52 @@ class FinancialAidActionView(UpdateAPIView):
     lookup_field = "id"
     lookup_url_kwarg = "financial_aid_id"
     queryset = FinancialAid.objects.all()
+
+
+class GetLearnerPriceForCourseView(APIView):
+    """
+    View for retrieving a leaner's price for a course run
+    """
+    authentication_classes = (SessionAuthentication, )
+    permission_classes = (IsAuthenticated, )
+
+    def get(self, request, *args, **kwargs):
+        """
+        Get request for GetLearnerPriceForCourseView
+        """
+        learner = request.user
+        program = get_object_or_404(
+            Program,
+            id=self.kwargs["program_id"],
+            live=True,
+            financial_aid_availability=True
```

Comment:
Can we remove this restraint here? @giocalitri and I were talking and it would be nice to use this functionality for the non financial aid programs too.